src/Compilers/Core/Portable/Generated/Operations.Generated.cs

```diff
@@ -3989,50 +3991,37 @@ internal EmptyOperation(SemanticModel? semanticModel, SyntaxNode syntax, bool is
-    internal abstract partial class BaseReturnOperation : OperationOld, IReturnOperation
-    {
-        internal BaseReturnOperation(OperationKind kind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ConstantValue constantValue, bool isImplicit)
-            : base(kind, semanticModel, syntax, type, constantValue, isImplicit) { }
-        public abstract IOperation ReturnedValue { get; }
-        public override IEnumerable<IOperation> Children
-        {
-            get
-            {
-                if (ReturnedValue is object) yield return ReturnedValue;
-            }
-        }
-        public override void Accept(OperationVisitor visitor) => visitor.VisitReturn(this);
-        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument) => visitor.VisitReturn(this, argument);
-    }
-    internal sealed partial class ReturnOperation : BaseReturnOperation, IReturnOperation
+    #nullable enable
+    internal sealed partial class ReturnOperation : Operation, IReturnOperation
     {
-        internal ReturnOperation(IOperation returnedValue, OperationKind kind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ConstantValue constantValue, bool isImplicit)
-            : base(kind, semanticModel, syntax, type, constantValue, isImplicit)
+        private IEnumerable<IOperation>? _lazyChildren;
+        internal ReturnOperation(IOperation? returnedValue, OperationKind kind, SemanticModel? semanticModel, SyntaxNode syntax, bool isImplicit)
+            : base(semanticModel, syntax, isImplicit)
         {
             ReturnedValue = SetParentOperation(returnedValue, this);
+            Kind = kind;
         }
-        public override IOperation ReturnedValue { get; }
-    }
-    internal abstract partial class LazyReturnOperation : BaseReturnOperation, IReturnOperation
-    {
-        private IOperation _lazyReturnedValue = s_unset;
-        internal LazyReturnOperation(OperationKind kind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, ConstantValue constantValue, bool isImplicit)
-            : base(kind, semanticModel, syntax, type, constantValue, isImplicit){ }
-        protected abstract IOperation CreateReturnedValue();
-        public override IOperation ReturnedValue
+        public IOperation? ReturnedValue { get; }
+        public override IEnumerable<IOperation> Children
         {
             get
             {
-                if (_lazyReturnedValue == s_unset)
+                if (_lazyChildren is null)
                 {
-                    IOperation returnedValue = CreateReturnedValue();
-                    SetParentOperation(returnedValue, this);
-                    Interlocked.CompareExchange(ref _lazyReturnedValue, returnedValue, s_unset);
+                    var builder = ArrayBuilder<IOperation>.GetInstance(1);
+                    if (ReturnedValue is not null) builder.Add(ReturnedValue);
```

Comment:
Why allocate a builder in the case it's `null`?

Reply:
Generation simplicity.

Reply:
I can add a prototype comment in the next PR to try and make the implementation of these properties more complex (and optimal). Did you have any other comments @jaredpar?